tour_model_eval/evaluation_pipeline.py

```diff
@@ -0,0 +1,173 @@
+
+                curr_score = gs.get_score(homo_second, percentage_second)
+                if curr_score not in tune_score:
+                    tune_score[curr_score] = (low, dist_pct, homo_second, percentage_second)
+
+        best_score = max(tune_score)
+        sel_tradeoffs = tune_score[best_score][0:2]
+    else:
+        sel_tradeoffs = (0,0)
+
+    return sel_tradeoffs
+
+
+def test(data,radius,low,dist_pct):
+    sim, bins, bin_trips, filter_trips = first_round(data, radius)
+    # it is possible that we don't have common trips for tuning or testing
+    # bins contain common trips indices
+    if len(bins) is not 0:
+        gs.compare_trip_orders(bins, bin_trips, filter_trips)
+        first_labels = get_first_label(bins)
+        # new_labels temporary stores the labels from the first round, but later the labels in new_labels will be
+        # updated with the labels after two rounds of clustering.
+        new_labels = first_labels.copy()
+        first_label_set = list(set(first_labels))
+        track = get_track(bins, first_labels)
+        # get request percentage for the subset for the first round
+        percentage_first = grp.get_req_pct(new_labels, track, filter_trips, sim)
+        # get homogeneity score for the subset for the first round
+        homo_first = gs.score(bin_trips, first_labels)
+        percentage_second, homo_second = second_round(first_label_set, first_labels, bin_trips, filter_trips, low,
+                                                      dist_pct, sim, new_labels, track)
+    else:
+        percentage_first = 1
+        homo_first = 1
+        percentage_second = 1
+        homo_second = 1
+    scores = gs.get_score(homo_second, percentage_second)
+    return homo_first,percentage_first,homo_second,percentage_second,scores
+
+
+def main(uuid = None):
+    user = uuid
+    radius = 100
+    trips = preprocess.read_data(user)
+    filter_trips = preprocess.filter_data(trips, radius)
+    tune_idx, test_idx = preprocess.split_data(filter_trips)
+    tune_data = preprocess.get_subdata(filter_trips, test_idx)
+    test_data = preprocess.get_subdata(filter_trips, tune_idx)
+    pct_collect_first = []
+    homo_collect_first = []
+    pct_collect_second = []
+    homo_collect_second = []
+    coll_score = []
+    coll_tradeoffs = []
+
+    # tune data
+    for j in range(len(tune_data)):
+        tuning_parameters = tune(tune_data[j], radius)
+        coll_tradeoffs.append(tuning_parameters)
+
+    # testing
+    for k in range(len(test_data)):
+        tradoffs = coll_tradeoffs[k]
+        low = tradoffs[0]
+        dist_pct = tradoffs[1]
+        homo_first, percentage_first, homo_second, percentage_second, scores = test(tune_data[k],radius,low,dist_pct)
+        pct_collect_first.append(percentage_first)
+        homo_collect_first.append(homo_first)
+        pct_collect_second.append(percentage_second)
+        homo_collect_second.append(homo_second)
+        coll_score.append(scores)
```

Comment:
then this will be significantly simplified.

Reply:
These are deleted.